### PR TITLE
[4128] Add data migration to fix funding methods for 2019/20

### PR DESCRIPTION
**Pull request by AbigailMcP**
### Context

https://trello.com/c/1POTbi8l/4128-fix-funding-rules-for-2019-20

### Changes proposed in this pull request

Add data migration that:
- Deletes the funding method's we have for 2019/20
- Adds the correct ones based on the section "Undergraduate bursary" in the manual: https://www.gov.uk/government/publications/initial-teacher-training-itt-bursary-funding-manual/initial-teacher-training-bursaries-funding-manual-2019-to-2020-academic-year#undergraduate-bursary

### Important business

- [ ] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [ ] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

**Comment by saliceti** [comment]
[Register: build workflow notification bug](https://trello.com/c/QTYqpCFJ/582-register-build-workflow-notification-bug)

**Comment by AbigailMcP** [comment]
[Fix funding rules for 2019/20](https://trello.com/c/1POTbi8l/4128-fix-funding-rules-for-2019-20)

File: db/data/20220516105534_fix_funding_methods_for20192020.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        AllocationSubjects::MATHEMATICS,
+        AllocationSubjects::PHYSICS,
+        AllocationSubjects::COMPUTING,
+        AllocationSubjects::MODERN_LANGUAGES,
+      ],
+    ),
+  ].freeze
+
+  def up
+    academic_cycle = AcademicCycle.find_by(start_date: "2019-09-01")
```
Comment (gpeng) [inline db/data/20220516105534_fix_funding_methods_for20192020.rb:26]: @stevehook has got a migration PR on the go to change these so we'll need to be careful with the order of merging if we use the start date. Maybe better to get them all and use the one with the 2019 start year 🤷

**Review by gpeng — CHANGES_REQUESTED** [review]

File: db/data/20220516105534_fix_funding_methods_for20192020.rb (hunk 1 above)
Comment (AbigailMcP) [reply db/data/20220516105534_fix_funding_methods_for20192020.rb]: Updated to use `for_year` method which doesn't depend on exact start date.

**Review by gpeng — APPROVED** [review]

**Pull request merged**